searx/engines/mastodon.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Mastodon is an open source alternative to large social media platforms
+like Twitter/X, Facebook, ...
+
+Since it's federated and self-hostable, there's a large amount of available
+instances, which can be chosen instead by modifying ``base_url``.
+
+We use their official API for searching, but unfortunately, their Search API
+forbids pagination without OAuth.
+
+That's why we use tootfinder.ch for finding posts, which doesn't support searching
+for users, accounts or other types of content on Mastodon however.
+"""
+
+from urllib.parse import urlencode
+from datetime import datetime
+
+about = {
+    "website": 'https://joinmastodon.org/',
+    "wikidata_id": 'Q27986619',
+    "official_api_documentation": 'https://docs.joinmastodon.org/api/',
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": 'JSON',
+}
+categories = ['social media']
+
+base_url = "https://mastodon.social"
+mastodon_type = "accounts"
+
+number_of_results = 10
```

Comment:
We should consider renaming this var because it is also used for getting the total number of results.

https://github.com/search?q=repo%3Asearxng%2Fsearxng%20number_of_results&type=code

Reply:
That's true, we should probably rename that to "page_size" for all engines (in a different PR).

Reply:
> That's true, we should probably rename that to "page_size" for all engines (in a different PR).

Thats what I also have thought often :) .. but we shouldn't:  ``number_of_results`` could be used in the `/etc/searxng/settings.yml` .. we don't know it .. currently we have monkey patching for all the settings of an engine .. may be one day we can establish a config scheme .. that scales .. and is downward compatible :)